components/oss/registry.go

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2021 Layotto Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package oss
+
+import (
+	"fmt"
+
+	"mosn.io/layotto/components/pkg/info"
+)
+
+type OssRegistry interface {
+	Register(fs ...*OssFactory)
+	Create(name string) (Oss, error)
+}
+
+type OssFactory struct {
```

Comment:
```suggestion
type Factory struct {
```
Since their is already a `oss` in the package path, how about naming it without `Oss`. 
Like what we did for other components:
![image](https://user-images.githubusercontent.com/26001097/180355362-d29f23c8-4a1b-48cd-b576-b9d8612874ca.png)